diffusion/train.py

```diff
@@ -38,6 +39,8 @@ def train(config: DictConfig) -> None:
         config.dataset.train_dataset,
         batch_size=config.dataset.train_batch_size // dist.get_world_size(),
     )
+    # Need to sleep for a bit to avoid dataloader crash
+    time.sleep(10)
```

Comment:
What's the reason to not use `dist.barrier`?

Reply:
This is just a hacky workaround until the upstream issue is fixed. I went with a `sleep` rather than a barrier after discussing with @knighton who suggested that this might be due to the shared memory state init not being done, and streaming's dataset has a barrier near the end of its init already.  In my testing adding a barrier seems to also work, but one has to handle calling `initialize_dist` within `train.py` to do it which winds up being kind of strange.

Reply:
Thanks for the explanation!